src/sensai/data_transformation.py

```diff
@@ -24,9 +25,14 @@ class DataFrameTransformer(ABC):
     def __init__(self):
         self._name = f"{self.__class__.__name__}-{id(self)}"
         self.__isFitted = False
-        self._changesColumns = False
-        self.__removedColumns: Optional[Set[str]] = None
-        self.__addedColumns: Optional[Set[str]] = None
+        self._dfHistory: Optional[DataFrameHistoryTracker] = None
+
+    # for backwards compatibility with persisted DFTs based on code prior to commit 7088cbbe
+    # They lack the __isFitted attribute and we assume that each such DFT was fitted
+    def __setstate__(self, d):
```

Comment:
I removed the double underscores for this reason, see comment above. I could use `super().__setstate__` (Pycharm complained for some reason), but setting `__dict__` also works. We actually do that at one place already:

![image](https://user-images.githubusercontent.com/35432522/104826965-84e70b80-5859-11eb-984f-44e983276b88.png)

Reply:
@opcode81 
Concerning
>super().__setstate__ is the most robust variant, because setting self.__dict__ (which is the default implementation in object) is not necessarily OK if we later introduce a super-class.

Pycharm was right to complain: when I replace it with `super().__setstate__(d)` the code does not run. 
```
    def __setstate__(self, d):
        d["_name"] = d.get("_name", f"{self.__class__.__name__}-{id(self)}")
        d["_isFitted"] = d.get("_isFitted", True)
        d["_dfHistory"] = d.get("_dfHistory", None)
>       super().__setstate__(d)
E       AttributeError: 'super' object has no attribute '__setstate__'
```
It seems like `ABC` is not derived from `object`, I didn't event know that was possible. Or there is some other black magic going on... Generally, it seems that in abstract classes `__setstate__` has to be implemented with the `__dict__` like I did it. Or am I missing something here?

Reply:
Not even `object` implements `__setstate__`. It's broken, which is unbelievable, really. The only classes I ever implemented `__setstate__` for were subclasses of `torch.nn.Module` (which implements `__setstate__`), so I didn't notice that `object` does not.
What this means is that when you implement `__setstate__` somewhere, you need to:
1. check if the super-class implements it and call it if so.
2. check if sub-classes implement it and if so change their implementations to call `super().__setstate__`.
This is insane! Everything would be totally "safe" if `object` implemented `__setstate__`; who's going to remember to do step 2?

Anyway, it's important to set the whole `__dict__` somehow (either via a call to super or explicitly), which you didn't previously, because you'd forget to update `__setstate__` when adding new properties.